### PR TITLE
Append-only wallet transactions

### PR DESCRIPTION
**Pull request by kjagiello**
Wallet transactions should never be altered once created. Should solve #19.

### CODE REVIEW

File: src/wallet/signals/handlers.py [hunk 1]
```diff
@@ -7,11 +7,5 @@
...
     wallet_obj = instance.wallet
     if created and instance.countable:
-        # A transaction has been created. If the transaction is finalized,
-        # we simply add it to the wallet balance.
         wallet_obj.balance += instance.signed_amount
-    elif not created:
-        # A transaction has been updated, so we simply recalculate
-        # the wallet balance.
-        wallet_obj.balance = wallet_obj.transactions.balance()
     wallet_obj.save()
```
Comment (flaeppe) [inline src/wallet/signals/handlers.py]: Do we always need to call `save()` here?

Comment (kjagiello) [inline src/wallet/signals/handlers.py]: True, true

File: src/wallet/migrations/0004_auto_20160309_1721.py [hunk 2]
```diff
@@ -13,14 +25,13 @@ def transfer_currency(apps, schema_editor):
     for wallet in Wallet.objects.all():
         incoming = WalletTrx.objects.filter(
             wallet=wallet.id,
-            trx_status=enums.TrxStatus.FINALIZED,
-            trx_type=enums.TrxType.INCOMING
+            trx_status=TrxStatus.FINALIZED,
```
Comment (kjagiello) [inline src/wallet/migrations/0004_auto_20160309_1721.py]: Incoming transactions that are PENDING do not contribute to the wallet balance.

File: src/wallet/api.py [hunk 3]
```diff
@@ -52,39 +55,40 @@ def get_transactions_by_ref(reference):
...
 
 @transaction.atomic
-def withdraw(owner_id, amount, reference=None):
+def withdraw(owner_id, amount, reference=None, pending=False):
     """Withdraw given amount from the wallet.
 
     Throw InsufficientFunds if there is not enough money in the wallet.
     """
-    assert amount.amount >= 0, "Cannot withdraw a negative amount of money"
+    assert amount.amount > 0, "The amount must be non-negative."
```
Comment (kjagiello) [inline src/wallet/api.py]: Fix incoming soon!

File: src/wallet/signals/handlers.py (hunk 1 above)
Comment (kjagiello) [inline src/wallet/signals/handlers.py]: No, definitely not!

**Pull request merged**